src/components/application_manager/test/commands/mobile/put_file_test.cc

```diff
@@ -0,0 +1,144 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+#include <string>
+
+#include "gtest/gtest.h"
+
+#include "interfaces/MOBILE_API.h"
+#include "application_manager/commands/commands_test.h"
+#include "application_manager/commands/command_request_test.h"
+#include "mobile/put_file_response.h"
+#include "mobile/put_file_request.h"
+#include "utils/make_shared.h"
+#include "smart_objects/smart_object.h"
+
+#include "application_manager/mock_application.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace put_file {
+
+using namespace application_manager;
+
+using ::testing::Return;
+using ::testing::_;
+
+namespace {
+const int32_t kConnectionKey = 1;
+}
+
+class PutFilesRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {};
+
+class PutFileResponseTest : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  void SetUp() OVERRIDE {
+    message_ = utils::MakeShared<SmartObject>(::smart_objects::SmartType_Map);
+    (*message_)[strings::msg_params] =
+        ::smart_objects::SmartObject(::smart_objects::SmartType_Map);
+
+    command_sptr_ =
+        CreateCommand<application_manager::commands::PutFileResponse>(message_);
+  }
+
+  MessageSharedPtr message_;
+  utils::SharedPtr<commands::PutFileResponse> command_sptr_;
+};
+
+TEST_F(PutFileResponseTest, Run_InvalidApp_ApplicationNotRegisteredResponce) {
+  ::smart_objects::SmartObject& message_ref = *message_;
+
+  message_ref[strings::params][strings::connection_key] = kConnectionKey;
+
+  utils::SharedPtr<Application> null_application_sptr;
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(null_application_sptr));
+  EXPECT_CALL(
+      app_mngr_,
+      SendMessageToMobile(
+          MobileResultCodeIs(mobile_api::Result::APPLICATION_NOT_REGISTERED),
+          _));
+  command_sptr_->Run();
+}
+
+TEST_F(PutFileResponseTest, Run_ApplicationRegistered_Success) {
+  ::smart_objects::SmartObject& message_ref = *message_;
+
+  message_ref[strings::params][strings::connection_key] = kConnectionKey;
+  message_ref[strings::msg_params][strings::success] = true;
+
+  utils::SharedPtr<Application> application_sptr =
+      utils::MakeShared<MockApplication>();
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(application_sptr));
+  EXPECT_CALL(
+      app_mngr_,
+      SendMessageToMobile(MobileResultCodeIs(mobile_apis::Result::SUCCESS), _));
+  command_sptr_->Run();
+}
+
+TEST_F(PutFilesRequestTest, Run_TooManyHmiNone_UNSUCCESS) {
+  MockAppPtr app(CreateMockApp());
+  SharedPtr<commands::PutFileRequest> command(
+      CreateCommand<commands::PutFileRequest>());
+
+  ON_CALL(app_mngr_, application(_)).WillByDefault(Return(app));
```

Comment:
@AGritsevich I have seen few lines above:
```
EXPECT_CALL(app_mngr_, application(_)).WillOnce(Return(mock_app_));
```

Please prefere at lease one style of expectations